cpp/test/sg/fil_test.cu

```diff
@@ -613,8 +614,11 @@ class TreeliteFilTest : public BaseFilTest {
     params.output_class = (ps.output & fil::output_t::CLASS) != 0;
     params.storage_type = storage_type;
     params.blocks_per_sm = ps.blocks_per_sm;
-    fil::from_treelite(handle, pforest, (ModelHandle)model.get(), &params);
+    char* model_shape_str = nullptr;
+    fil::from_treelite(handle, pforest, (ModelHandle)model.get(), &params,
+                       ps.print_model_shape, &model_shape_str);
     CUDA_CHECK(cudaStreamSynchronize(stream));
+    if (ps.print_model_shape) printf("%s", model_shape_str);
```

Comment:
It's better not to print anything here. Instead, you can do something similar to what you do in the Python test: check that the resulting string contains the substrings it is expected to contain.